ports.js

```diff
@@ -1,51 +1,231 @@
 /* eslint-disable no-console */
 /* globals encrypt:false, decrypt:false */
+const CONTENT_KEY = "pad";
 
-let app;
-if (typeof Elm === 'undefined') {
-  // This happens if we're in the context of Electron.
-  const Elm = require('./hoverpad.js');
-  app = Elm.Main.fullscreen();
-} else {
-  app = Elm.Main.fullscreen();
+const IS_WEB_EXTENSION = (typeof chrome === "object" && typeof chrome.storage === "object");
+
+function storePassphrase(passphrase) {
+  if (!IS_WEB_EXTENSION) {
+    // Insecurely storing the passphrase.
+    sessionStorage.setItem("temporaryPassphrase", btoa(passphrase));
+    return Promise.resolve(null);
+  } else {
+    return setItem("temporaryPassphrase", passphrase);
+  }
+}
+
+function dropPassphrase() {
+  if (!IS_WEB_EXTENSION) {
+    sessionStorage.removeItem("temporaryPassphrase");
+    return Promise.resolve(null);
+  } else {
+    return setItem("temporaryPassphrase", null);
+  }
 }
 
-app.ports.decryptData.subscribe(function(data) {
-  console.log(data);
-  if (!data.content) {
-    app.ports.dataDecrypted.send(null);
-    return;
-  }
-  decrypt(data.passphrase, data.content)
-    .then(content => {
-      app.ports.dataDecrypted.send(content);
-    })
-    .catch(err => {
-      console.error('Error decrypting', err);
-      app.ports.dataNotDecrypted.send(err.message);
+function getPassphrase() {
+  // Insecurely retrieving the passphrase.
+  if (!IS_WEB_EXTENSION) {
+    const passphrase = sessionStorage.getItem("temporaryPassphrase");
+    if (passphrase === null) {
+      return null;
```

Comment:
Should this return `null` directly, or wrap it in a Promise object, a la `return Promise.resolve(null);`?